### PR TITLE
[structured authz][kubeadam] avoid setting default authorization-mode when structured authz is wanted

### LINKED ISSUE #123651: [kubeadam][structured authz] can't use structured authz due to default `--authorization-mode` flag
### What happened?

**TL;DR**
https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L232 prevents us from using structured authz (setting `--authorization-config` flag) as those flags are mutually exclusive

/cc @palnabarun (as the author of https://github.com/kubernetes/kubernetes/pull/120154)
/cc @liggitt 

FYI @stealthybox @robscott 

To use structured authz you need to define the authorization configuration using the `--authorization-config` flag.

However this flag is mutually exclusive with `--authorization-modes` and `--authorization-webhook-*` flags.  (refer to [1](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/pkg/kubeapiserver/options/authorization.go#L113) & [2](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#configuring-the-api-server-using-an-authorization-config-file))

Kubeadam adds `--authorization-mode` by default ([1](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L232) & [2](https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/cmd/kubeadm/app/phases/controlplane/manifests.go#L241)) which prevents us from using `--authorization-config`

### What did you expect to happen?

I should be able to define the `authorization-config`  flag and have structured authz working.

### How can we reproduce it (as minimally and precisely as possible)?

Spin up a kind cluster using this kind config (make sure the images are > 1.29 for structured authz to be avaliable)

```yaml
kind: Cluster
apiVersion: kind.x-k8s.io/v1alpha4
featureGates:
  StructuredAuthorizationConfiguration: true
kubeadmConfigPatches:
  - |
    kind: ClusterConfiguration
    metadata:
      name: config
    apiServer:
      extraArgs:
        authorization-config: "/files/authorization_config.yaml" 
      extraVolumes:
      - name: api-server-basic-auth-files
        hostPath: "/files"
        mountPath: "/files"
        readOnly: true
nodes:
- role: control-plane
  # add a mount from /path/to/my/files on the host to /files on the node
  extraMounts:
  - hostPath: <some-path-on-my-host>/refgrant-poc/files
    containerPath: /files
```

(For this bug, the content of `/files/authorization_config.yaml` doesn't matter.)

However, the cluster become unhealthy.
docker exec to the kind cluster container, where we can see the inner container of the apiserver has exited. 
Running `crictl logs <container>` we can see this error

image

### Anything else we need to know?

_No response_

### Kubernetes version

<details>

```console
The kind cluster version image: `kindest/node:v1.29.2`
```

</details>

### Cloud provider

<details>
N/A -I am using kind
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

### PR DESCRIPTION
**Pull request by LiorLieberman**
do not set authorization-mode in api server when authorization-config is provided

#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
This PR enabled kubeadam users to use structured authz

#### Which issue(s) this PR fixes:

Fixes #123651 

#### Does this PR introduce a user-facing change?

```release-note
[kubeadam][structured authz] avoid setting default `--authorization-mode` when `--authorization-config` is provided
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

/cc @liggitt 
FYI @robscott @stealthybox

### CODE REVIEW

**Comment by liggitt** [comment]
Good catch, thanks

/lgtm

**Comment by LiorLieberman** [comment]
@liggitt thanks for quickly lgtm. I added another test after you lgtmed, mind re-lgtm it?

**Comment by liggitt** [comment]
/lgtm

will still need an approve from kubeadm reviewer

File: cmd/kubeadm/app/phases/controlplane/manifests_test.go [hunk 1]
```diff
@@ -514,6 +514,103 @@ func TestGetAPIServerCommand(t *testing.T) {
...
+				"--authorization-config=/path/to/authorization/config/file",
+				"--advertise-address=1.2.3.4",
+				fmt.Sprintf("--etcd-servers=https://127.0.0.1:%d", kubeadmconstants.EtcdListenClientPort),
+				"--etcd-cafile=" + filepath.Join(testCertsDir, "etcd/ca.crt"),
+				"--etcd-certfile=" + filepath.Join(testCertsDir, "apiserver-etcd-client.crt"),
+				"--etcd-keyfile=" + filepath.Join(testCertsDir, "apiserver-etcd-client.key"),
+			},
+		},
+		{
+			// Note that we do not block it at this level but api server would fail to start.
```
Comment (neolit123) [inline cmd/kubeadm/app/phases/controlplane/manifests_test.go:563]: SGTM given the mutex is mentioned in the new flag description.
https://github.com/kubernetes/kubernetes/blob/7c11cc9cfcdc54a7ca4efdc10609bf421bfa54d4/pkg/kubeapiserver/options/authorization.go#L190

**Review by neolit123 — COMMENTED** [review]
thanks
/approve

**Review by neolit123 — COMMENTED** [review]
@LiorLieberman 
should this be backported to 1.29.x?

**Comment by liggitt** [comment]
> @LiorLieberman
> should this be backported to 1.29.x?

generally I wouldn't expect backports for alpha features

**Pull request merged**